src/WCSimConstructCylinder.cc

```diff
@@ -71,6 +71,9 @@ G4LogicalVolume* WCSimDetectorConstruction::ConstructCylinder()
   totalAngle  = 2.0*pi*rad*(WCBarrelRingNPhi*WCPMTperCellHorizontal/WCBarrelNumPMTHorizontal) ;
   // angle per regular cell:
   dPhi        =  totalAngle/ WCBarrelRingNPhi;
+  barrelPhiOffset = 0.;
```

Comment:
Define in WCSimDetectorConfigs.cc instead?
Can also define relevant macro commands and make it useful for all geometries

Reply:
Except when zero, the value of `barrelPhiOffset` is calculated from (actually just equal to) the value of `dPhi` which is calculated in the line above, not available in WCSimDetectorConfigs.cc. So unless I also moved all the existing calculations of all `dPhi` the other variables it depends on (`totalAngle`, `WCBarrelRingNPhi`), it has to be set here. So I just tried to follow the same procedure as these other variables that are defined here. I don't like the explicit check for the WCTE geometry, but there's already existing similar checks on lines 1458 and 2964 (which are actaully for a related reason to the need for the barrel realignment: the WCTE geometry has the end-cap mPMTs aligned with a central row/column rather than an even number of rows/columns either side of the centre) so I just did the same here. I did not include checking for the older WCTE geometry so that if anyone wanted to use the older geometry for some reason, it would be unchanged from older studies. We could change all the places where it explicitly checks the WCTE geometry(s), so other geometried could use it, but with WCSim development being effectively abandoned now for everything except WCTE, and the geometry code frankly needs a complete rethink anyway, I think that's probably beyond the scope of this PR.

Reply:
I would probably still want a simple macro commands like:
-------------------------------------------
BarrelRotationType [none dPhiUnit dPhiExtraUnit custom]
BarrelRotationUnit [double]
-------------------------------------------
In this way `barrelPhiOffset` can be calculated dynamically, and let this feature useful for other geometries (or even GHOST)

Reply:
If adding a macro then I'd rather just make it a boolean for whether it's rotated half a tower or not. I can see some possible benefit of that so we can still build the old wcte geometry for comparison purposes. The WCTE support structure design is finalised and I don't see a reason now that we'd want to rotate by a value other than half a tower, since that would always lead to misalignment of the endcap rows and columns with the barrel towers, which would potentially also cause overlap issues that we'd need to check for. It's generally bad coding practice to overgeneralise (which WCSim already does, trying to build a generic detector that then needs tweaking for each specific design) and then add all these options to fix it, unless or until we actually need to use these options. If someone wants to actually need to rotate the detector barrel by random values for some reason, then that functionality could be added as part of that work. Having seen how bad the WCSim geometry code is, I think for GHOST we would want to rewrite it to avoid these problems entirely, so the other option would be a complete rewrite in discussion with GHOST developers and following their coding standards, but I don't think we have time for that while needing this fixed now for WCTE in WCSim.

Reply:
This could work. Making a simple macro command is better than hard-coded the change.

Reply:
I've added the macro command. Will test in a few days (cluster where I got visualisation working for testing is currently down for a few more days of maintenance)